app/models/agents/data_output_agent.rb

```diff
@@ -261,19 +264,22 @@ def receive_web_request(params, method, format)
       interpolate_with('events' => source_events) do
         items = source_events.map do |event|
           interpolated = interpolate_options(options['template']['item'], event)
-          interpolated['guid'] = {'_attributes' => {'isPermaLink' => 'false'},
-                                  '_contents' => interpolated['guid'].presence || event.id}
-          date_string = interpolated['pubDate'].to_s
-          date =
-            begin
-              Time.zone.parse(date_string)  # may return nil
-            rescue => e
-              error "Error parsing a \"pubDate\" value \"#{date_string}\": #{e.message}"
-              nil
-            end || event.created_at
-          interpolated['pubDate'] = date.rfc2822.to_s
-          interpolated
+          if !params['category_filter'].present? || (params['category_filter'] == interpolated['category']) 
```

Comment:
This could be a guard to make the diff cleaner

```ruby
next nil if params['category_filter'].present? && params['category_filter'] != interpolated['category']
```